lib/galaxy/tool_util/lint.py

```diff
@@ -344,10 +366,26 @@ def lint_tool_source_with(lint_context, tool_source, extra_modules=None) -> Lint
                         lint_context.lint(name, value, tool_xml)
                 else:
                     lint_context.lint(name, value, tool_source)
+            elif inspect.isclass(value) and issubclass(value, Linter) and not inspect.isabstract(value):
+                lint_context.lint(name, value.lint, tool_source)
     return lint_context
 
 
 def lint_xml_with(lint_context, tool_xml, extra_modules=None) -> LintContext:
     extra_modules = extra_modules or []
     tool_source = get_tool_source(xml_tree=tool_xml)
     return lint_tool_source_with(lint_context, tool_source, extra_modules=extra_modules)
+
+
+def list_linters(extra_modules: Optional[List[str]] = None) -> List[str]:
```

Comment:
I think that's actually a rare case where you can use a Metaclass:

```ipython
In [1]: linters = {}
   ...:
   ...: class LinterMeta(type):
   ...:     def __new__(cls, clsname, bases, attrs):
   ...:         newclass = super(LinterMeta, cls).__new__(cls, clsname, bases, attrs)
   ...:         linters[clsname] = newclass
   ...:         return newclass
   ...:
   ...:
   ...: class MyLinter(metaclass=LinterMeta):
   ...:     pass
   ...:
   ...: linters
Out[1]: {'MyLinter': __main__.MyLinter}
```

Reply:
Thanks for pointing me to metaclasses. Never took the time to learn about them. 

Still I hope that https://github.com/galaxyproject/galaxy/pull/17081/commits/3a98a3a2628839a476c6a9c934098852c64bb774 is sufficient?